### ISSUE TITLE
POJO copy constructor should use interface when codegen of <interfaces/> = true

### ISSUE THREAD

**Issue by anuraaga** [jOOQ/jOOQ#7055]
### Expected behavior and actual behavior:

The copy constructor of a POJO uses interface methods when interfaces are enabled, e.g.

```java
public class User implements IUser

  private final ULong id;
  private final String name;
  private final String phoneNumber;

  public User(IUser value) {
    this.id = value.getId();
    this.name = value.getName();
    this.phoneNumber = value.getPhoneNumber();
  }
}
```

This would allow the copy constructor to work both on POJOs and records, which is especially useful when using `immutablePojos` as there is no `Record.into()` generated for such a case. 

If user code is like the currently only supporte case, where only POJOs are copied to POJOs, there should be no difference in performance as the compiler will inline everything with no polymorphic overhead. 

### Steps to reproduce the problem:

Enable POJOs and interfaces in codegen.

### Versions:

- jOOQ: 3.10.3

**Comment by lukaseder** [comment]
Thanks for the suggestion. You're right, that would definitely make sense.

**Comment by lukaseder** [comment]
This is implemented for 3.11.0 - it was rather easy to do.

The change is not binary compatible, which is why we won't merge it to 3.10.0. Given that source compatibility is fulfulled, we're not going to overload the constructor for backwards-compatibility.

Thanks again for the suggestion.
